src/server/service/match.ts

```diff
@@ -235,17 +244,23 @@ export class Match {
   }
 
   // TODO: make a proper DB relation with user and matches instead of doing this
+  // TODO: rename to get player stats
   private async getPlayerPreviousMatches() {
     const promises = this.players
       .filter((player) => !player.isBot)
       .map(async (player) => {
         if (this._playerPreviousMatches.get(player.userId)) return;
+        if (this._playerAchievements.get(player.userId)) return;
 
         const matchRooms = (await selectMatchPlayedByUser(player.userId)).map(
           (match) => match.match.room,
         );
-
         this._playerPreviousMatches.set(player.userId, matchRooms);
+
+        this._playerAchievements.set(
+          player.userId,
+          await selectUserAchievements(player.userId),
```

Comment:
`async` calls as params break legibility and can lead to unpredictable behaviour, please call this before setting the value in the map